rfd/0010-api.md

```diff
@@ -35,89 +33,154 @@ func connectClient() (*auth.Client, error) {
 
     // TLS client is the only client
     return auth.NewTLSClient(clientConfig)
+}
 ```
 
 ## Details
 
-### Approach
+A new `api` package will be made these new features and any existing Teleport Auth client logic, most of which is is currently in `teleport/lib/auth/clt.go`. The new package will only support Teleport's gRPC client which will gradually replace the deprecated HTTP client. Other logic such as the web client and certificate reading logic will also be moved to the `api` package. 
+
+The `api` package will be made a Go module, with no dependency on Teleport libraries. This will make it easy to import the API client with a small import footprint. Users will be able to run `go get github.com/gravitational/teleport/api` to get the latest version of the client and get started.
```

Comment:
Yes, that's a good point. I think we've pretty much settled on keeping the api a go sub module since the main repo has direct dependencies on it. It would be a pain to develop in separate repositories since most changes require both modules to be changed. The more likely long term solution in my mind is to turn more teleport packages that aren't purely internal (e.g. `lib/utils`, `lib/defaults`) into submodules. I've seen a few large scale projects like https://github.com/hashicorp/vault use this approach. This will be difficult due to the convoluted package dependency graph within the teleport module though.